parquet/src/main/java/org/apache/iceberg/data/parquet/BaseParquetReaders.java

```diff
@@ -76,6 +67,31 @@ protected ParquetValueReader<T> createReader(
   protected abstract ParquetValueReader<T> createStructReader(
       List<Type> types, List<ParquetValueReader<?>> fieldReaders, Types.StructType structType);
 
+  protected ParquetValueReaders.PrimitiveReader<?> fixedReader(ColumnDescriptor desc) {
+    return null;
+  }
+
+  protected ParquetValueReaders.PrimitiveReader<?> int96Reader(ColumnDescriptor desc) {
+    return null;
+  }
+
+  protected Optional<ParquetValueReader<?>> dateReader(ColumnDescriptor desc) {
+    return Optional.empty();
+  }
+
+  protected Optional<ParquetValueReader<?>> timeReader(ColumnDescriptor desc, TimeUnit unit) {
+    return Optional.empty();
+  }
+
+  protected Optional<ParquetValueReader<?>> timestampReader(
+      ColumnDescriptor desc, TimeUnit unit, boolean isAdjustedToUTC) {
+    return Optional.empty();
+  }
+
+  protected Optional<ParquetValueReader<?>> uuidReader(ColumnDescriptor desc) {
```

Comment:
Why is there a factory method for UUID? Both uses of this class have the same representation so a new method is not required.

Reply:
UUID is unsupported by generic so far. I got an error. 

I can keep it local to this file if we don't need factory method. Some has factory method and some are local to class. It is little confusing, I will take a step back and analyze, when it has to be factory and when local.

Reply:
I see. So the Generic writers and readers never handled UUID !

Reply:
CI just failed for `ArrowReaderTest > testReadAll()`. 

Generic reader and writer expects byte[] based UUID handling and internal reader and writer works directly on UUID object. So, we do need a factory method here as we need to supply different kinds of UUID representation.